modules/nf-core/gawk/main.nf

```diff
@@ -0,0 +1,40 @@
+process GAWK {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "anaconda::gawk=5.1.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/gawk:5.1.0' :
+        'quay.io/biocontainers/gawk:5.1.0' }"
+
+    input:
+    tuple val(meta), path(input)
+    path(program_file)
+
+    output:
+    tuple val(meta), path("${prefix}")  , emit: output
+    path "versions.yml"                 , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def args2 = task.ext.args2 ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}.${input.getExtension}" // This default is up to debate, please post about it on Slack or open a PR with your suggested change
+
+    program = program_file ? "-f ${program_file}" : "${args2}"
```

Comment:
On a Linux box, the man page says
> gawk [ POSIX or GNU style options ] -f program-file [ -- ] file ...
> gawk [ POSIX or GNU style options ] [ -- ] program-text file ...

On a Mac:
> awk [ -F fs ] [ -v var=value ] [ 'prog' | -f progfile ] [ file ...  ]

So I'm not sure we need `$args2` at all ? `$args` can encompass all the options

Reply:
I would like to keep the option to specify the program inside of the configuration file. This is especially handy for small logic statements IMO. `args2` is only used to specify a program if no program file has been given

Reply:
I'll try and make that a bit more clear

Reply:
Aaah I get what you mean. Since the tool will mostly be ran in a Linux based container I think this will make things unnecessarily complicated IMO